go/oasis-test-runner/scenario/e2e/gas_fees_staking.go

```diff
@@ -65,7 +65,11 @@ func (sc *gasFeesImpl) Fixture() (*oasis.NetworkFixture, error) {
 	}
 
 	var tee node.TEEHardware
-	err = tee.FromString(sc.TEEHardware)
+	teeStr, err := sc.flags.GetString(cfgTEEHardware)
+	if err != nil {
```

Comment:
Huh, now we need to handle errors in all places where fetching flags? Does that mean that if you pass an invalid type (e.g., a string where an integer value is expected), the error will only be emitted once it hits the `Get*` call? Can you check?

It would be nice if this could be simplified.

Reply:
Actually all setter functions for flags are called when setting up scenarios (when generating individual parameter sets), so if you pass invalid value (e.g. string instead of int), it will correctly complain.

Reply:
> It would be nice if this could be simplified.

The idea was that flags are now explicit to separate them from other non-settable scenario fields. Before they were just wrappers around scenario variables and initialization/usage was unclear.

Reply:
I added `getTEEHardware()` helper which does this now.

Reply:
> Actually all setter functions for flags are called when setting up scenarios (when generating individual parameter sets), so if you pass invalid value (e.g. string instead of int), it will correctly complain.

So when can this actually return an error?

Reply:
> So when can this (e.g., the `Get*` methods) actually return an error?

If a programmer registers a flag as fs.String(cfgMyFlag) and then he calls GetInt(cfgMyFlag). The new `getTEEHardware()` helper returns error, if TEE flag is invalid (it's not either `""` or `"intel-sgx"`).

Reply:
This should be a panic (programmer error) not a runtime error.

Reply:
We'll just ignore those for now as `viper` does.